compiler/executable/match_/planner/plan.rs

```diff
@@ -128,22 +135,51 @@ fn make_builder<'a>(
                     variable_registry,
                     expressions,
                     statistics,
+                    per_call_costs,
                 )
                 .with_inputs(negation.conjunction().captured_variables(block_context))
                 .plan(),
             ),
             NestedPattern::Optional(_) => todo!(),
         }
     }
-
-    let mut plan_builder = ConjunctionPlanBuilder::new(type_annotations, statistics);
+    // Compute variables which must be bound from a parent scope.
```

Comment:
I think this is a useful functionality that can be moved to the BlockContext, or the Conjunction itself, depending on where we have similar functionalities?